### PR TITLE
[WIP] Continue deeplink after install application

### PR DESCRIPTION
**Pull request by Programistich**
**Background**

Please, describe here why you create this PR

**Changes**

Here what change in this PR..

**Test plan**

..and how we can test it

### CODE REVIEW

File: components/deeplink/api/src/main/java/com/flipperdevices/deeplink/api/DeepLinkParser.kt [hunk 1]
```diff
@@ -15,5 +15,5 @@ interface DeepLinkParser {
      * and second is a throwable if deeplink is null
```
Comment (LionZXY) [inline components/deeplink/api/src/main/java/com/flipperdevices/deeplink/api/DeepLinkParser.kt:15]: Please, fix javadoc also if it possible

File: components/share/receive/src/main/java/com/flipperdevices/share/receive/ReferrerReceiver.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+class ReferrerReceiver : BroadcastReceiver() {
+
+    @Inject
+    lateinit var singleActivityApi: SingleActivityApi
+
+    @Inject
+    lateinit var deepLinkParser: DeepLinkParser
+
+    override fun onReceive(context: Context?, intent: Intent?) {
```
Comment (LionZXY) [inline components/share/receive/src/main/java/com/flipperdevices/share/receive/ReferrerReceiver.kt:23]: `BroadcastReceiver` can't pass null `Context`

File: components/share/receive/src/main/java/com/flipperdevices/share/receive/ReferrerReceiver.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        val uri = intent?.data
```
Comment (LionZXY) [inline components/share/receive/src/main/java/com/flipperdevices/share/receive/ReferrerReceiver.kt:24]: NIT: You can simplify it to `val uri = intent?.data ?: return`

File: components/share/receive/src/main/java/com/flipperdevices/share/receive/ReferrerReceiver.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        if (uri != null && context != null) {
+            GlobalScope.launch {
```
Comment (LionZXY) [inline components/share/receive/src/main/java/com/flipperdevices/share/receive/ReferrerReceiver.kt:26]: Can you, please, specify dispatcher? For example, `Dispatcher.IO`

File: components/share/receive/src/main/java/com/flipperdevices/share/receive/ReferrerReceiver.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class ReferrerReceiver : BroadcastReceiver() {
+
+    @Inject
+    lateinit var singleActivityApi: SingleActivityApi
+
+    @Inject
+    lateinit var deepLinkParser: DeepLinkParser
```
Comment (LionZXY) [inline components/share/receive/src/main/java/com/flipperdevices/share/receive/ReferrerReceiver.kt:21]: Critical: How this object injected?

**Review by LionZXY — CHANGES_REQUESTED** [review]

File: components/deeplink/api/src/main/java/com/flipperdevices/deeplink/api/DeepLinkParser.kt (hunk 1 above)
Comment (Programistich) [reply components/deeplink/api/src/main/java/com/flipperdevices/deeplink/api/DeepLinkParser.kt:15]: Fix in Javadoc why I put context but not activity?

**Comment by LionZXY** [comment]
Close for now

**Pull request closed without merging**